### ISSUE TITLE
About Arcface training strategy

### ISSUE THREAD

**Issue by tangchen2** [deepinsight/insightface#887]
im trying to train insightFace on CASIA - Webface, and i tryed to train arcface from scratch, but it seems difficult, although the accuracy could be high, but the validation rate on lfw is very low, likely acc 70% but validation rate just 5%.
could anyone give me some advice on how to train arcface? 
BTW, when i train triplet loss, i also encoutner the same problem, accuracy could be high , but validation rate is very low, Am i wrong somewhere? hope someone could help me, thanks in advance.

**Comment by Retrospection** [comment]
> im trying to train insightFace on CASIA - Webface, and i tryed to train arcface from scratch, but it seems difficult, although the accuracy could be high, but the validation rate on lfw is very low, likely acc 70% but validation rate just 5%.
> could anyone give me some advice on how to train arcface?
> BTW, when i train triplet loss, i also encoutner the same problem, accuracy could be high , but validation rate is very low, Am i wrong somewhere? hope someone could help me, thanks in advance.

Maybe you need to align the dataset first

**Comment by tangchen2** [comment]
> > im trying to train insightFace on CASIA - Webface, and i tryed to train arcface from scratch, but it seems difficult, although the accuracy could be high, but the validation rate on lfw is very low, likely acc 70% but validation rate just 5%.
> > could anyone give me some advice on how to train arcface?
> > BTW, when i train triplet loss, i also encoutner the same problem, accuracy could be high , but validation rate is very low, Am i wrong somewhere? hope someone could help me, thanks in advance.
> 
> @tangchen2 Maybe you need to align the dataset first

Sorry , i forgot to mention about that, i have aligned the dataset to (112, 112)size before training

**Comment by Retrospection** [comment]
> > > im trying to train insightFace on CASIA - Webface, and i tryed to train arcface from scratch, but it seems difficult, although the accuracy could be high, but the validation rate on lfw is very low, likely acc 70% but validation rate just 5%.
> > > could anyone give me some advice on how to train arcface?
> > > BTW, when i train triplet loss, i also encoutner the same problem, accuracy could be high , but validation rate is very low, Am i wrong somewhere? hope someone could help me, thanks in advance.
> > 
> > 
> > @tangchen2 Maybe you need to align the dataset first
> 
> Sorry , i forgot to mention about that, i have aligned the dataset to (112, 112)size before training

In my experience, the most possible reason of your bad validation acc is that you may do a wrong face alignment or your batch size is too small. It's better for you to try large batch size for example 512, or carefully check your face alignment process.

**Comment by tangchen2** [comment]
yes, you are right, i changed to use the datazoo provided by the author instead of doing it myself and the performance got better. I think i must did something wrong when i crop the dataset, thank you so much! 
And ome more question, did you know about the triplet loss training, i first trained the model with softmax loss and the accuracy reached over 98% , validation rate reach around 95% , and i want to use triplet loss to get higher accuracy, but the truth is that when is use triplet loss to train the softmax loss model, the validation rate and accuracy decrease rapidly, i have tried different lr from 0.01 to 1e-7, it seems doesnt work, any idea about that problem? Thank you in advance!

**Comment by Retrospection** [comment]
> i changed to use the datazoo provided by the author instead of doing it m

sorry, but have no experience in fine-tuning on triplet loss..........................

**Comment by tangchen2** [comment]
> > i changed to use the datazoo provided by the author instead of doing it m
> 
> sorry, but have no experience in fine-tuning on triplet loss..........................

okay, thanks anyway, in my opinion, triplet loss is really hard to finetune...

**Comment by Vietdung113** [comment]
> yes, you are right, i changed to use the datazoo provided by the author instead of doing it myself and the performance got better. I think i must did something wrong when i crop the dataset, thank you so much!
> And ome more question, did you know about the triplet loss training, i first trained the model with softmax loss and the accuracy reached over 98% , validation rate reach around 95% , and i want to use triplet loss to get higher accuracy, but the truth is that when is use triplet loss to train the softmax loss model, the validation rate and accuracy decrease rapidly, i have tried different lr from 0.01 to 1e-7, it seems doesnt work, any idea about that problem? Thank you in advance!

Can you provided your training parameter, or some method used to on training phase.
I downloaded dataset from datazoo but it still got low accuracy

**Comment by He-haitao** [comment]
why my train accuracy is so low while veryfication accuracy is so high? train datasest is faces_umd, veryfication datatest is LFW,

![image](https://user-images.githubusercontent.com/41290343/71781017-c5843600-3004-11ea-8509-0148b44c1e8d.png)

Can you give me a  hand？ Thanks in advance.

**Comment by nttstar** [comment]
Softmax accuracy will increase rapidly after learning rate decreased.

**Comment by He-haitao** [comment]
想问一下训练断开了，如何继续训练啊，我用那个预训练模型的参数，config.py里面直接写pretrained= 'model’，那个pre_trainedepoch也要自己设置的吗？还要记住之前的epoch是多少？是这样的么
![image](https://user-images.githubusercontent.com/41290343/72116405-22754880-3385-11ea-9373-2af9c8e91ab5.png)
但是为啥说没有这个文件呢？这个pretrained输入格式是什么样的呢？

**Comment by nanhui69** [comment]
> Softmax accuracy will increase rapidly after learning rate decreased.

for the train script, is this train_softmax.py which under “src/ ” floder?

**Comment by nanhui69** [comment]
> im trying to train insightFace on CASIA - Webface, and i tryed to train arcface from scratch, but it seems difficult, although the accuracy could be high, but the validation rate on lfw is very low, likely acc 70% but validation rate just 5%.
> could anyone give me some advice on how to train arcface?
> BTW, when i train triplet loss, i also encoutner the same problem, accuracy could be high , but validation rate is very low, Am i wrong somewhere? hope someone could help me, thanks in advance.

could you provide the whole model train process ? or share you social tool for communication。。。。

**Comment by nanhui69** [comment]
> > > > im trying to train insightFace on CASIA - Webface, and i tryed to train arcface from scratch, but it seems difficult, although the accuracy could be high, but the validation rate on lfw is very low, likely acc 70% but validation rate just 5%.
> > > > could anyone give me some advice on how to train arcface?
> > > > BTW, when i train triplet loss, i also encoutner the same problem, accuracy could be high , but validation rate is very low, Am i wrong somewhere? hope someone could help me, thanks in advance.
> > > 
> > > 
> > > @tangchen2 Maybe you need to align the dataset first
> > 
> > 
> > Sorry , i forgot to mention about that, i have aligned the dataset to (112, 112)size before training
> 
> In my experience, the most possible reason of your bad validation acc is that you may do a wrong face alignment or your batch size is too small. It's better for you to try large batch size for example 512, or carefully check your face alignment process.
hey，i have try to train arcface model from scratch , and i have finish it ,but some error happen that dont save model when train model completely ,what’s wrong is it?
